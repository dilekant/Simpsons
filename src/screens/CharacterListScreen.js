import React, {useContext, useEffect, useLayoutEffect, useState} from 'react';
import {View, Text, StyleSheet} from 'react-native';

const CharacterListScreen = ({navigation}) => {

  useEffect(() => {
    getData();
  }, []);

  const getData = () => {
    /* let url = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons`;
        axios
            .get(url)
            .then((response) => {
                //setData(response.data.data);
                console.log(response.data.data);
            })
            .catch((error) => {
                console.log(error);
            }); */
  };

  return (
    <View style={styles.container}>
      <Text>Character List Screen</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default CharacterListScreen;
