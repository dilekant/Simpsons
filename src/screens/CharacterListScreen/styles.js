import {StyleSheet} from 'react-native';
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  addContainer: {
    width: wp('100%'),
    position: 'absolute',
    alignItems: 'center',
    bottom: hp('2%'),
  },
  addButton: {
    backgroundColor: '#258DD0',
    width: hp('6.5%'),
    height: hp('6.5%'),
    borderRadius: hp('6.5%') / 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  footer: {
    height: hp('10.5%'),
  },
});

export default styles;
