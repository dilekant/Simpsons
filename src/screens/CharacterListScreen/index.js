import React, {useContext, useEffect, useState} from 'react';
import {
  View,
  FlatList,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';
import {heightPercentageToDP as hp} from 'react-native-responsive-screen';
import ListItem from '../../components/ListItem';
import Add from '../../icons/Add';
import {RefreshContext} from '../../contexts/RefreshContext';
import styles from './styles';

const CharacterListScreen = ({navigation}) => {
  const {shouldRefresh, refreshProduct} = useContext(RefreshContext);
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    getSimpsons();
  }, []);

  useEffect(() => {
    const refreshListener = navigation.addListener('focus', () => {
      if (refreshProduct) {
        onRefresh(true);
        shouldRefresh(false);
      }
    });
    return refreshListener;
  }, [navigation, refreshProduct]);

  const onRefresh = async () => {
    await getSimpsons();
  };

  const getSimpsons = async () => {
    setLoading(true);
    try {
      const value = await AsyncStorage.getItem('SIMPSONS');
      if (value !== null) {
        setData(JSON.parse(value));
        setLoading(false);
      } else {
        getData();
      }
    } catch (e) {
      // error reading value
      setLoading(false);
    }
  };

  const getData = () => {
    let url = `https://5fc9346b2af77700165ae514.mockapi.io/simpsons`;
    axios
      .get(url)
      .then(response => {
        setData(response.data);
        setSimpsons(response.data);
        setLoading(false);
      })
      .catch(error => {
        console.log(error);
        setLoading(false);
      });
  };

  const setSimpsons = async response => {
    console.log('response setle');
    try {
      const jsonValue = JSON.stringify(response);
      await AsyncStorage.setItem('SIMPSONS', jsonValue);
    } catch (e) {
      // saving error
    }
  };

  const deletePress = item => {
    let simpsonsArray = data.filter(e => {
      return e !== item;
    });
    setData(simpsonsArray);
    setSimpsons(simpsonsArray);
  };

  return (
    <View style={styles.container}>
      {loading ? (
        <View style={styles.loadingContainer}>
          <ActivityIndicator size={'large'} color={'#000000'} />
        </View>
      ) : (
        <FlatList
          data={data}
          style={styles.listContainer}
          numColumns={1}
          showsVerticalScrollIndicator={false}
          keyExtractor={(item, index) => index.toString()}
          renderItem={({item, index}) => (
            <ListItem
              {...item}
              key={index}
              onPress={() =>
                navigation.navigate('CharacterDetailScreen', {detail: item})
              }
              onPressDelete={() => deletePress(item)}
            />
          )}
          ListFooterComponent={<View style={styles.footer} />}
        />
      )}
      <View style={styles.addContainer}>
        <TouchableOpacity
          onPress={() => navigation.navigate('AddCharacterScreen')}
          style={styles.addButton}>
          <Add stroke={'#ffffff'} width={hp('3.5%')} height={hp('3.5%')} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default CharacterListScreen;
