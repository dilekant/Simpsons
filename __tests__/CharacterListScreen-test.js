import React from 'react';
import renderer from 'react-test-renderer';
import CharacterListScreen from '../src/screens/CharacterListScreen';

test('renders correctly', () => {
  const tree = renderer.create(<CharacterListScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});