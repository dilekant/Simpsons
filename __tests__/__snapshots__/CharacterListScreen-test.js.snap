// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={
      <View
        style={
          Object {
            "height": 140.07,
          }
        }
      />
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "height": 140.07,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 26.68,
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3084E0",
          "borderRadius": 43.355000000000004,
          "height": 86.71000000000001,
          "justifyContent": "center",
          "opacity": 1,
          "width": 86.71000000000001,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={46.690000000000005}
        bbWidth={46.690000000000005}
        className="feather feather-plus"
        fill="none"
        focusable={false}
        height={46.690000000000005}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#ffffff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 46,
              "width": 46,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={46.690000000000005}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4294967295}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M12 5v14M5 12h14"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
